--- cdo-1.7.0rc5/src/Makefile	2015-09-09 11:59:34.958385035 +0200
+++ src_original/src/Makefile	2015-09-09 11:57:00.931344449 +0200
@@ -774,6 +774,22 @@
 	@rm -f cdo$(EXEEXT)
 	$(AM_V_CCLD)$(cdo_LINK) $(cdo_OBJECTS) $(cdo_LDADD) $(LIBS)
 
+#############################################################
+#inserts by Ben van Werkhoven
+#############################################################
+MAIN_OBJ = cdo-cdo.$(OBJEXT)
+AMUSE_OBJS = $(filter-out $(MAIN_OBJ),$(cdo_OBJECTS))
+
+libcdo-cdolib.a:
+	echo "Building CDO library for AMUSE"
+	ar ruv libcdo-cdolib.a $(AMUSE_OBJS)
+	ranlib libcdo-cdolib.a
+
+
+#############################################################
+#end of inserts by Ben
+#############################################################
+
 cdotest$(EXEEXT): $(cdotest_OBJECTS) $(cdotest_DEPENDENCIES) $(EXTRA_cdotest_DEPENDENCIES) 
 	@rm -f cdotest$(EXEEXT)
 	$(AM_V_CCLD)$(cdotest_LINK) $(cdotest_OBJECTS) $(cdotest_LDADD) $(LIBS)
